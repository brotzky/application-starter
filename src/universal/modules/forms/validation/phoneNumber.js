import validator from 'validator';

export default function(value) {
  if (!value) { return undefined; }
  
  const str = value || '';
  return validator.isMobilePhone(str, 'en-CA')
    ? undefined
    : {
        defaultMessage: 'must be a valid phone number',
        id: 'form.errors.invalidPhoneNumber',
        values: {},
      };
}
