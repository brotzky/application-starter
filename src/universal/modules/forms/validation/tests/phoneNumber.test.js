import phoneNumber from '../phoneNumber';

describe('[Form Validation] phoneNumber', () => {
  describe('phoneNumber()', () => {
    it('should return undefined if nothing is passed it', () => {
      expect(phoneNumber()).toMatchSnapshot();
    });

    it('should return no errors if a valid phone number is passed in', () => {
      expect(phoneNumber('4169999999')).toMatchSnapshot();
    });

    it('should return an error when an invalid phone number is passed in', () => {
      expect(phoneNumber('123')).toMatchSnapshot();
    });
  });
});
