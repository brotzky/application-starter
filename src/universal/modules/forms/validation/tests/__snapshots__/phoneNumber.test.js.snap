// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Form Validation] phoneNumber phoneNumber() should return an error when an invalid phone number is passed in 1`] = `
Object {
  "defaultMessage": "must be a valid phone number",
  "id": "form.errors.invalidPhoneNumber",
  "values": Object {},
}
`;

exports[`[Form Validation] phoneNumber phoneNumber() should return no errors if a valid phone number is passed in 1`] = `undefined`;

exports[`[Form Validation] phoneNumber phoneNumber() should return undefined if nothing is passed it 1`] = `undefined`;
